lib/serializer.ts

```diff
@@ -393,10 +379,8 @@ function serializeDictionaryType(serializer: Serializer, mapper: DictionaryMappe
       `mapper and it must of type "object" in ${objectName}.`);
   }
   const tempDictionary: { [key: string]: any } = {};
-  for (const key in object) {
-    if (object.hasOwnProperty(key)) {
-      tempDictionary[key] = serializer.serialize(mapper.type.value, object[key], objectName);
-    }
+  for (const key of Object.keys(object)) {
+    tempDictionary[key] = serializer.serialize(mapper.type.value, object[key], objectName + "." + key);
```

Comment:
This lets us give better errors by telling which dictionary key failed serialization, e.g.

`myDictionary.firstName with value "Rikki&" should satisfy the constraint /[A-Za-z_]/`

instead of `myDictionary with value "Rikki&" should satisfy...`